src/backend/gl/src/device.rs

```diff
@@ -788,6 +810,48 @@ impl d::Device<B> for Device {
             vertex_buffers[vb.binding as usize] = Some(*vb);
         }
 
+        let mut uniforms = Vec::new();
+        {
+            let gl = &self.share.context;
+            let mut count = 0;
+            let mut uniform_max_size = 0;
+
+            gl.GetProgramiv(program, gl::ACTIVE_UNIFORMS, &mut count);
+            gl.GetProgramiv(
+                program,
+                gl::ACTIVE_UNIFORM_MAX_LENGTH,
+                &mut uniform_max_size,
+            );
+            for uniform in 0..count {
+                let mut length = 0;
+                let mut size = 0;
+                let mut utype = 0;
+                let mut name = Vec::with_capacity(uniform_max_size as usize);
+                name.set_len(uniform_max_size as usize - 1);
```

Comment:
Why a minus one?
> GL_ACTIVE_UNIFORM_MAX_LENGTH 
> params returns the length of the longest active uniform variable name for program, including the null termination character (i.e., the size of the character buffer required to store the longest uniform variable name). If no active uniform variables exist, 0 is returned.

Reply:
To not include the null terminator mostly.

Reply:
`glGetUniformLocation`'s string must be null terminated:
> name:
> Points to a null terminated string containing the name of the uniform variable whose location is to be queried.

Therefore, `name` must be null terminated. `set_len` should be given `uniform_max_size`, and the last byte should be manually set to '\0'.

`glGetActiveUniform` should still receive `uniform_max_size - 1`

Reply:
As it stands, `glGetActiveUniform` could write to all `0..uniform_max_size - 1` bytes, and then `glGetUniformLocation` would read past the buffer.